### PR TITLE
Set up basic routing and a client-facing test page

### PR DESCRIPTION
**Pull request by Implausibility**
This is the skeleton for the routing and a page to confirm that Express
is serving static files correctly. There's also different linter
settings in the server folder, to reflect that it's in the Node
require() environment.

Task: https://trello.com/c/434gn30f

### CODE REVIEW

File: server/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+  env: {
+    commonjs: true,
```
Comment (eLeiLei) [inline server/.eslintrc.js:3]: Hi @Implausibility, I see that you chose commonjs, airbnb-base and no react plugins. Are these settings working for your linter and React files?

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  <head>
+    <title>
+      Project Overnight
+    </title>
+  </head>
+  <body>
+    <div id="reservation"></div>
+    <script src="bundle.js"></script>
+  </body>
+</html>
```
Comment (eLeiLei) [inline public/index.html:11]: You've got a newline notification here.

File: server/.eslintrc.js (hunk 1 above)
Comment (Implausibility) [reply server/.eslintrc.js:3]: Those are my settings for inside the server folder, where everything is just Node. My general project linter files include the React plugins and the 'import' syntax.

File: public/index.html (hunk 2 above)
Comment (Implausibility) [reply public/index.html:11]: Thanks, I completely missed that.

File: server/.eslintrc.js (hunk 1 above)
Comment (eLeiLei) [reply server/.eslintrc.js:3]: Ok great. You did account for React.

**Pull request merged**